app/app/src/main/java/org/omecproject/up4/cli/FlowsStatsCommand.java

```diff
@@ -0,0 +1,108 @@
+/*
+ SPDX-License-Identifier: Apache-2.0
+ SPDX-FileCopyrightText: 2022-present Intel Corporation
+ */
+package org.omecproject.up4.cli;
+
+import com.google.common.collect.Maps;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.omecproject.up4.impl.Up4AdminService;
+import org.onlab.packet.Ip4Address;
+import org.onosproject.cli.AbstractShellCommand;
+import org.onosproject.net.behaviour.upf.UpfCounter;
+
+import java.util.Map;
+
+/**
+ * UP4 UE flow stats.
+ */
+@Service
+@Command(scope = "up4", name = "flow-stats",
+        description = "Read all UE flows stats")
+public class FlowsStatsCommand extends AbstractShellCommand {
+
+    private static final String SEPARATOR = "-".repeat(40);
+    private static final int SLEEP_TIME_MS = 200;
+
+    @Override
+    protected void doExecute() throws Exception {
+        Up4AdminService adminService = get(Up4AdminService.class);
+
+        Map<Ip4Address, UpfCounter> beforeDownlink = Maps.newHashMap();
+        adminService.getDownlinkFlows().forEach(dlUpfFlow -> {
+            beforeDownlink.put(dlUpfFlow.getSession().ueAddress(), dlUpfFlow.getCounter());
```

Comment:
I don't remember what we mean by downlink flow, but can there be multiple flow for the same session? Does it make sense to aggregate/sum all flows in the same counter value?

Reply:
Yes, we can have multiple session downlink flow per UE session. A downlink flow, it's basically a QoS Flow (a single application flow). We can aggregate per UE, however, this can be done only in the DL

Reply:
Ah ok, then downlink flow = downlink termination, right? Unless I'm missing something, it would be good to use consistent naming.

Reply:
About having stats per applications, would be good if we can make that an CLI argument. In most cases I think we only care about aggregate stats, so showing per-application might come later.

Reply:
OK, I'll postpone the per-app stats for a later PR. Tracking this in Jira in: https://jira.opennetworking.org/browse/SDFAB-1190.

About downlink flow, it aggregates all UpfEntities that are used to represent a UPF QoS Flow (I believe that's the right 3GPP term). Meaning it contains a session, termination, tunnel peer, counter, session meter and app meter.  The term downlink flow is being used for "historical" reason, and to differentiate with the "UpfTerminationDownlink" class that comes from UpfProgrammable API. I would avoid using DownlinkTermination as name, to avoid confusion with the UpfProgrammable class.